plaso/storage/redis/redis_store.py

```diff
@@ -0,0 +1,488 @@
+      identifier_string = event_identifier.decode('utf-8')
+      event_identifier = identifiers.RedisKeyIdentifier(identifier_string)
+      yield self._GetAttributeContainerByIdentifier(
+          self._CONTAINER_TYPE_EVENT, event_identifier)
+
+  def GetSerializedAttributeContainers(
+      self, container_type, cursor, maximum_number_of_items):
+    """Fetches serialized attribute containers.
+
+    Args:
+      container_type (str): attribute container type.
+      cursor (int): Redis cursor.
+      maximum_number_of_items (int): maximum number of containers to
+          retrieve, where 0 represent no limit.
+
+    Returns:
+      tuple: containing:
+        int: Redis cursor.
+        list[bytes]: serialized attribute containers.
+    """
+    name = self._GenerateRedisKey(container_type)
+    # Redis treats None as meaning "no limit", not 0.
+    if maximum_number_of_items == 0:
+      maximum_number_of_items = None
+
+    cursor, items = self._redis_client.hscan(
+        name, cursor=cursor, count=maximum_number_of_items)
+    return cursor, items
+
+  # pylint: disable=arguments-differ
+  def Open(self, redis_client=None, url=None):
+    """Opens the store.
+
+    Args:
+      redis_client (Optional[Redis]): Redis client to query. If specified, no
+          new client will be created. If no client is specified a new client
+          will be opened connected to the Redis instance specified by 'url'.
+      url (Optional[str]): URL for a Redis database. If not specified, the
+          DEFAULT_REDIS_URL will be used.
+
+    Raises:
+      IOError: if the store is already connected to a Redis instance.
+      OSError: if the store is already connected to a Redis instance.
+    """
+    if not url:
+      url = self.DEFAULT_REDIS_URL
+
+    if self._redis_client:
+      raise IOError('Redis client already connected')
+
+    if redis_client:
+      self._redis_client = redis_client
+    else:
+      self._redis_client = redis.from_url(url=url, socket_timeout=60)
+
+    client_name = self._GenerateRedisKey('')
+    self._SetClientName(self._redis_client, client_name)
+
+    metadata_key = self._GenerateRedisKey('metadata')
+    if not self._redis_client.exists(metadata_key):
+      self._WriteStorageMetadata()
+
+  def Remove(self):
+    """Removes the contents of the store from Redis."""
+    merging_key = '{0:s}-{1:s}'.format(
+        self._session_identifier, self._MERGING_KEY_NAME)
+    self._redis_client.hdel(merging_key, self._task_identifier)
+
+    sorted_event_key = self._GenerateRedisKey(self._EVENT_INDEX_NAME)
+    self._redis_client.delete(sorted_event_key)
+
+    task_completion_key = self._GenerateRedisKey(
+        self._CONTAINER_TYPE_TASK_COMPLETION)
+    self._redis_client.delete(task_completion_key)
+
+    metadata_key = self._GenerateRedisKey('metadata')
+    self._redis_client.delete(metadata_key)
+
+  @classmethod
+  def ScanForProcessedTasks(
+      cls, session_identifier, url=None, redis_client=None):
```

Comment:
nit, please reorder the kwargs alphabetically. Also see: https://github.com/log2timeline/l2tdocs/blob/master/process/Style-guide.md#function-and-method-arguments

Reply:
Done